redfish-core/lib/systems.hpp

```diff
@@ -2076,6 +2076,113 @@ inline void getEnabledPanelFunctions(
         });
 }
 
+/**
+ * Execute a Panel Enabled Function
+ */
+inline void
+    executePanelFunction(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                         const uint8_t funcNo)
+{
+    BMCWEB_LOG_DEBUG << "Execute Panel function " << std::to_string(funcNo);
+
+    crow::connections::systemBus->async_method_call(
+        [asyncResp,
+         funcNo](const boost::system::error_code& ec,
+                 const sdbusplus::message_t& msg,
+                 const std::tuple<bool, std::string, std::string>& result) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR << "Execute a panel function D-bus error:  "
+                             << ec.value();
+
+            const sd_bus_error* dbusError = msg.get_error();
+            if (dbusError == nullptr)
+            {
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            if (dbusError->name ==
+                std::string_view("xyz.openbmc_project.Common.Error.NotAllowed"))
+            {
+                messages::operationNotAllowed(asyncResp->res);
+                return;
+            }
+            else if (dbusError->name ==
+                     std::string_view(
+                         "xyz.openbmc_project.Common.Error.InternalFailure"))
+            {
+                messages::operationFailed(asyncResp->res);
+                return;
+            }
+            messages::internalError(asyncResp->res);
```

Comment:
You want to add a ERROR trace here?

Reply:
To print dbusError because it is a value we were not expecting

Reply:
I guess you already have it above 

           BMCWEB_LOG_ERROR << "Execute a panel function D-bus error:  "
                             << ec.value();

Reply:
Rearranged the code to add ERROR to show operationFailed/NotAllowed cases.